Added Spring animations

kwiedenhaupt opened the pull request:
I wanted to try out the spring animation basics. 
1.  Downloaded the react-spring package.

Added 3 animations.
1.  fade-in/out of ExpandableContainer
2. fade-in/out better transition for tabs on SkillRatingsComponent
3. Cascading appearance of goals in GoalsSectionComponent -- this may be annoying lol. just wanted to try it.  I can reverse the commit!

iamnotreddy reviewed (changes requested):
Awesome job implementing a brand new library! I think you would have found this extremely challenging a month ago - really happy to see you progress to this level :)

couple thoughts on the animations themselves:
- not sure i'm feeling the use of trail on the goals component, when you minimize the section you can see the previous render for a couple seconds. I know this is the intention but the look doesn't feel clean enough for me and it's kind of disorienting
- love the spring animation between switching tabs on skillRatings... this feels super smooth, let's definitely keep this!
- is the spring height working on the ExpandableContainer? It doesn't appear to be. Ideally the expandable containers slide open to their expanded height when clicking plus, slide closed to their closed height when clicking minus. Can we try to implement it that way?

kwiedenhaupt commented:
1.  I deleted the useTrail on the Goals feature.  I totally agree with you.
2. I totally revamped the animation on the ExpandableContainer and added a useHeight function in the API folder to calculate the height. -- let me know what you think about the smoothness, etc.
3. Also let me know what you think about the template animation.

vithushan19 reviewed (commented):
awesome work on getting this going! great to see you learning a new frameworks and applying them.
